### PR TITLE
Numba version and deprecation updates

### PR DESCRIPTION
**Pull request by gmarkall**
Some small Numba-related changes:

- Testing on CI appears to be picking up Numba 0.53.1, and there have been quite a lot of CUDA changes between Numba 0.49 (the old minimum) and 0.53.1 - increase the minimum required Numba version to 0.53.1
  accordingly.
- Remove old import guards and alternatives for Numba versions < 0.49 - these were needed because of Numba's internals refactor between 0.48 and 0.49.
- Replace the use of `inspect_ptx()` with `compile_ptx()` in `test_ptx_generic()` - `inspect_ptx()` has always had various issues so it is being deprecated and will be removed in a later version. `compile_ptx()` provides a better alternative. See https://github.com/numba/numba/pull/6953

### CODE REVIEW

**Review by kkraus14 — COMMENTED** [review]
Changes LGTM but we need to handle the versioning in the conda environment files and the conda recipes as well.

**Comment by gmarkall** [comment]
> Changes LGTM but we need to handle the versioning in the conda environment files and the conda recipes as well.

Many thanks for catching this!

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
Need to update the integration repo as well, I will put in a PR

https://github.com/rapidsai/integration/blob/branch-0.19/conda/recipes/versions.yaml#L105-L106

**Review by ajschmidt8 — APPROVED** [review]

File: python/cudf/requirements/cuda-11.1/dev_requirements.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ hypothesis
 mimesis
 mypy==0.782
 nbsphinx
-numba>=0.49.0,!=0.51.0
+numba>=>=0.53.1
```
Comment (galipremsagar) [inline python/cudf/requirements/cuda-11.1/dev_requirements.txt:19]:
```suggestion
numba>=0.53.1
```

looks like a typo here

**Review by galipremsagar — CHANGES_REQUESTED** [review]

**Review by galipremsagar — APPROVED** [review]

**Comment by kkraus14** [comment]
@gpucibot merge

**Comment by kkraus14** [comment]
@gpucibot merge

**Pull request merged**